api/src/controllers/record-utils.js

```diff
@@ -72,44 +73,21 @@ const addError = (record, error) => {
 };
 
 /*
- * Try to parse SMSSync gateway sent_timestamp field and use it for
- * reported_date.  Particularly useful when re-importing data from gateway to
+ * Try to parse sent_timestamp field and use it for reported_date.
+ * Particularly useful when re-importing data from gateway to
  * maintain accurate reported_date field.
  *
  * return unix timestamp integer or undefined
  */
 const parseSentTimestamp = str => {
-
   if (typeof str === 'number') {
     str = String(str);
   } else if (typeof str !== 'string') {
     return;
   }
 
-  // smssync 1.1.9 format
-  const match1 = str.match(/(\d{1,2})-(\d{1,2})-(\d{2})\s(\d{1,2}):(\d{2})(:(\d{2}))?/);
-
-  if (match1) {
-    const ret = new Date();
-
-    let year = ret.getFullYear();
-    year -= year % 100; // round to nearest 100
-    ret.setYear(year + parseInt(match1[3], 10)); // works until 2100
-
-    ret.setMonth(parseInt(match1[1],10) - 1);
-    ret.setDate(parseInt(match1[2], 10));
-    ret.setHours(parseInt(match1[4], 10));
-    ret.setMinutes(match1[5]);
-    ret.setSeconds(match1[7] || 0);
-    ret.setMilliseconds(0);
-    return ret.valueOf();
-  }
-
-  // smssync 2.0 format (ms since epoch)
-  const match2 = str.match(/(\d{13,})/);
-
-  if (match2) {
-    const ret = new Date(Number(match2[1]));
+  if (DATE_NUMBER_STRING.test(str)) {
+    const ret = new Date(Number(str));
```

Comment:
Previously this would extract numbers from strings such as `fishcake-1525808599351-window` and correctly convert them to dates, but now will convert them into `Invalid Date` objects.

Reply:
Ugh, you're right. That's so ugly. I'll put it back.